### PR TITLE
(feat) Routing Rules - Implementation for routing rules for ingest.

### PR DESCRIPTION
**Pull request by mdhaman**
SD-1548

### CODE REVIEW

File: server/apps/rules/routing_rules.py [hunk 1]
```diff
@@ -69,10 +72,10 @@ class RoutingRuleSchemeResource(Resource):
                             'allowed': ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT']
```
Comment (amagdas) [inline server/apps/rules/routing_rules.py]: This list should be declared only once and reused.

**Comment by sivakuna-aap** [comment]
:+1:

**Comment by amagdas** [comment]
Rebase needed.

File: server/apps/rules/routing_rule_validator.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import logging
+import re
+
+logger = logging.getLogger(__name__)
+
+
+class RoutingRuleValidator():
+
+    def is_valid_rule(self, item, rule_filter):
+        if not filter:
```
Comment (petrjasek) [inline server/apps/rules/routing_rule_validator.py]: seems like it should be `rule_filter`, `filter` is always defined

**Comment by petrjasek** [comment]
:+1: but would be good to catch some specific exceptions in those try/catch blocks

File: server/apps/rules/routing_rule_validator.py (hunk 2 above)
Comment (sivakuna-aap) [inline server/apps/rules/routing_rule_validator.py]: `filter` is not mandatory property. When `filter` is not specified it the rule will match all the ingested articles associated with an ingested provider. For example, a rule can be something like this All articles from AAP with an email id should go to email desk.

Comment (mdhaman) [inline server/apps/rules/routing_rule_validator.py]: You are right ```filter``` is always defined. It should be ```rule_filter```.  

@petrjasek  which exceptions are you referring?

File: server/apps/rules/routing_rules.py [hunk 3]
```diff
@@ -155,20 +181,35 @@ def __validate_routing_scheme(self, routing_scheme):
...
+        if schedule:
+            day_of_week = [str(week_day).upper() for week_day in schedule.get('day_of_week', [])]
+            if not (len(set(day_of_week) & set(self.day_of_week)) == len(day_of_week)):
+                return False
+
+            if schedule.get('hour_of_day_from') or schedule.get('hour_of_day_to'):
+                try:
+                    return datetime.strptime(schedule.get('hour_of_day_from'), '%H%M') < \
+                        datetime.strptime(schedule.get('hour_of_day_to'), '%H%M')
+                except:
```
Comment (petrjasek) [inline server/apps/rules/routing_rules.py]: exceptions like this

Comment (mdhaman) [inline server/apps/rules/routing_rules.py]: done..

**Comment by petrjasek** [comment]
:+1:

**Comment by amagdas** [comment]
:+1:

**Comment by mugurrus** [comment]
:+1:

**Pull request merged**